src/dehydrated_devices.rs

```diff
@@ -33,18 +78,54 @@ impl DehydratedDevices {
     #[wasm_bindgen]
     pub async fn rehydrate(
         &self,
-        pickle_key: &Uint8Array,
+        pickle_key: &DehydratedDeviceKey,
         device_id: &DeviceId,
         device_data: &str,
     ) -> Result<RehydratedDevice, JsError> {
-        let pickle_key: [u8; 32] =
-            pickle_key.to_vec().try_into().map_err(|_| JsError::new("Wrong key length"))?;
+        let sdk_pickle_key =
+            SdkDehydratedDeviceKey::from_slice(pickle_key.inner.to_vec().as_slice())?;
+
         Ok(self
             .inner
-            .rehydrate(&pickle_key, &device_id.inner, serde_json::from_str(device_data)?)
+            .rehydrate(&sdk_pickle_key, &device_id.inner, serde_json::from_str(device_data)?)
             .await?
             .into())
     }
+
+    /// Get the cached dehydrated device pickle key if any.
+    ///
+    /// None if the key was not previously cached (via
+    /// [`Self::save_dehydrated_device_pickle_key`]).
+    ///
+    /// Should be used to periodically rotate the dehydrated device to avoid
+    /// OTK exhaustion and accumulation of to_device messages.
+    #[wasm_bindgen(js_name = "getDehydratedDeviceKey")]
+    pub async fn get_dehydrated_device_key(&self) -> Result<Option<DehydratedDeviceKey>, JsError> {
+        let key = self.inner.get_dehydrated_device_pickle_key().await?;
+        Ok(key.map(DehydratedDeviceKey::from))
+    }
+
+    /// Store the dehydrated device pickle key in the crypto store.
+    ///
+    /// This is useful if the client wants to periodically rotate dehydrated
+    /// devices to avoid OTK exhaustion and accumulated to_device problems.
```

Comment:
I think I need a bit more explanation about how to use this. Do I just make a new `DehydratedDeviceKey` and call `save_dehydrated_device_key` with it?

Reply:
Maybe I should remove these comments if they are bringing confusion?

The rotation of dehydrated device is another topic not handled by this code. But caching the key and re-use it is important if you want to rotate without having to ask for the 4S key every time (in order to update 4S with a new dehydrated device key for the next login to use)

Reply:
hrm; I think it would be *nice* to have documentation of how to correctly rotate a dehydrated device. But it's not essential, and yes, this comment is probably bringing more confusion than help.

On balance: yes, probably best to remove this comment.